components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java

```diff
@@ -126,7 +127,28 @@ public Response addRegistry(RegistryDTO body, MessageContext messageContext) {
     @Override
     public Response createRegistryEntry(String registryId, RegistryEntryDTO registryEntry, InputStream
             definitionFileInputStream, Attachment definitionFileDetail, MessageContext messageContext) {
-        return Response.ok().entity(registryEntry).build();
+        EndpointRegistry registryProvider = new EndpointRegistryImpl();
+        EndpointRegistryEntry createdEntry = null;
+        try {
+            EndpointRegistryInfo endpointRegistry = registryProvider.getEndpointRegistryByUUID(registryId);
+            if (endpointRegistry == null) {
+                RestApiUtil.handleResourceNotFoundError("Endpoint registry with the id: " + registryId +
+                        " is not found", log);
+            }
+            ResourceFile definitionFile = new ResourceFile(definitionFileInputStream, definitionFileDetail
```

Comment:
do we store the content type in the db? do we need it for any purpose?

Reply:
No, it is not stored. Up to now, we haven't use the content-type field. Added ResourcFile instead of InputStream for definition file, in case it is needed in the future

Reply:
does this content-type mean xml, json, yaml etc?